src/transformations/io-transformations/PrintfToCoutTransformation.cpp

```diff
@@ -0,0 +1,224 @@
+#include "../../../include/transformations/io-transformations/PrintfToCoutTransformation.h"
+#include "../../../include/transformations/io-transformations/IOUtils.h"
+
+using clang::IfStmt, clang::CallExpr;
+using clang::Lexer, clang::CharSourceRange, clang::SourceLocation;
+using llvm::isa, llvm::cast, llvm::errs;
+using std::stoi, std::vector, std::string, std::regex_search,
+std::regex, std::sregex_iterator, std::distance;
+
+// ------------ PrintfToCoutVisitor ------------
+
+PrintfToCoutVisitor::PrintfToCoutVisitor(Rewriter * rewriter) :
+        rewriter(rewriter), sm(rewriter->getSourceMgr()), opt(rewriter->getLangOpts()) {}
+
+bool PrintfToCoutVisitor::VisitCallExpr(CallExpr *e) {
+    auto srcloc = e->getBeginLoc();
+    // Collecting printf or std::printf expressions
+    if (sm.isWrittenInMainFile(srcloc)) {
+        auto const srctext = getAsText(e);
+        if (srctext == "std::printf" || srctext == "printf") {
+            printfExpressions.push_back(e);
+        }
+    }
+    return true;
+}
+
+bool PrintfToCoutVisitor::VisitUsingDirectiveDecl(UsingDirectiveDecl *ud) {
+    auto srcloc = ud->getBeginLoc();
+    // Collecting all used namespaces
+    if (sm.isWrittenInMainFile(srcloc)) {
+        usedNamespaces.push_back(ud);
+    }
+    return true;
+}
+
+bool PrintfToCoutVisitor::determineIfSimplePrintfCommand(const CallExpr * e) {
+    string formatstring = getAsText(e->getArg(0)).str();
+
+    // We rewrite simple printf commands directly.
+    // If the number of args matches with the number from the regex,
+    // we have a relative simple printf command
+    regex const re(regexPlaceholders);
+
+    int const match_count(distance(
+            sregex_iterator(formatstring.begin(), formatstring.end(), re),
+            sregex_iterator()));
+
+    return (match_count == (e->getNumArgs() - 1));
+}
+
+string PrintfToCoutVisitor::append_std(string word) {
+    for (auto ud : usedNamespaces) {
+        if (ud->getNominatedNamespace()->getQualifiedNameAsString() == word) {
+            return word;
+        }
+    }
+    return "std::"+ word;
```

Comment:
Add space before the plus operator.

P.S. It strange that linter doesn't point to this.